### PR TITLE
testing(run): add retry behavior to cloudrunci gcloud commands

### LINKED ISSUE #1740: Add retry to gcloud commands in Cloud Run end-to-end testing
Cloud Run flaky tests seems mostly the result of gcloud commands randomly failing. Modify cloudrunci to add retry.

### PR DESCRIPTION
**Pull request by grayside**
Fixes #1740

Flaky tests for Cloud Run seem like they may be addressed through simple retry. This PR modifies gcloud execution in cloudrunci to make 3 attempts with 2 second intervals to push a failing operation through.

This initial cut makes some tweaks to testutil retry to support callers without a testing object. I thought this would be preferable to duplicating retry code in the same module.

### CODE REVIEW

**Review by ks1707 — APPROVED** [review]

**Review by tbpg — APPROVED** [review]

**Comment by grayside** [comment]
The number of clicks to find the test failures seems to be growing...

TestGcloudRetry needs to be behind a system-test flag, gcloud is not installed by default.

Other test failures are unrelated to this PR.

File: internal/cloudrunci/gcloud_test.go [hunk 1]
```diff
@@ -27,21 +27,32 @@ func TestCreateIDToken(t *testing.T) {
...
 	}
 
 	want := "gcloud - manage Google Cloud Platform resources and developer workflow"
 	if got := string(out); !strings.Contains(got, want) {
 		t.Errorf("gcloud: got (%s), want (%s)", got, want)
 	}
 }
+
+func TestGcloudRetry(t *testing.T) {
+	_, err := gcloud("failing", exec.Command("command-does-not-exist"))
```
Comment (tbpg) [inline internal/cloudrunci/gcloud_test.go:50]: Can we hide the logs from this? It's a bit confusing to see "errors" in the logs when they're expected.

**Comment by tbpg** [comment]
Ignore the `do not merge` label shenanigans -- was testing a new bot.

@grayside, what do we need to get this merged?

**Comment by grayside** [comment]
Returning to this PR now.

File: internal/cloudrunci/gcloud_test.go (hunk 1 above)
Comment (grayside) [reply internal/cloudrunci/gcloud_test.go:56]: Setting the logs to ioutil.Discard for the duration of this test.

File: internal/cloudrunci/gcloud_test.go [hunk 2]
```diff
@@ -27,21 +29,36 @@ func TestCreateIDToken(t *testing.T) {
...
 	}
 
 	want := "gcloud - manage Google Cloud Platform resources and developer workflow"
 	if got := string(out); !strings.Contains(got, want) {
 		t.Errorf("gcloud: got (%s), want (%s)", got, want)
 	}
 }
+
+func TestGcloudRetry(t *testing.T) {
+	originalWriter := log.Writer()
```
Comment (tbpg) [inline internal/cloudrunci/gcloud_test.go:52]: This was added in Go 1.13, so this is failing to compile with Go 1.11.

https://golang.org/doc/go1.13#log

I think we can assume `os.Stderr`.

**Review by tbpg — CHANGES_REQUESTED** [review]

File: internal/cloudrunci/gcloud_test.go (hunk 2 above)
Comment (grayside) [reply internal/cloudrunci/gcloud_test.go:52]: Somehow I keep forgetting this limitation. Thanks for patiently flagging it each time.

**Review by tbpg — APPROVED** [review]

**Pull request merged**